django/db/models/fields/tuple_lookups.py

```diff
@@ -129,6 +130,19 @@ def as_sql(self, compiler, connection):
         return super().as_sql(compiler, connection)
 
 
+class TupleComparisonMixin:
+    def as_sql(self, compiler, connection):
+        if (
+            not connection.features.supports_tuple_comparison_in_subquery
+            and isinstance(self.rhs, Query)
+            and self.rhs.subquery
+        ):
+            lookup = self.lookup_name
+            msg = f"{lookup} for composite fields is not supported on this backend"
```

Comment:
Sorry I forgot to mention but the message should likely mention that it's only the case for subquery.